src/org/centos/pipeline/PackagePipelineUtils.groovy

```diff
@@ -91,6 +91,63 @@ def setMessageFields(String messageType, String artifact) {
     return [ 'topic': topic, 'properties': messagePropertiesString, 'content': messageContentString ]
 }
 
+/**
+ * Method that uses contra-lib shared library
+ * to create the ci.artifact.test.messageType messages
+ * @param messageType: queued, running, complete, error
+ * @param artifact: dist-git-pr, koji-build
+ * @return
+ */
+// Plan is to rename to setMessageFields and remove function above once everything seems fine and stable
+def setTestMessageFields(String messageType, String artifact) {
+    // Set values that go in multiple closures
+    myTopic = "${MAIN_TOPIC}.ci.${artifact}.test.${messageType}"
+    myType = (artifact == 'koji-build') ? 'tier0' : 'build'
+    myComponent = env.fed_repo
+    myIssuer = env.fed_owner ?: fed_username
+    // PR builds are scratch so defaulting to scratch = true
+    myScratch = env.isScratch ? env.isScratch.toBoolean() : true
+    taskid = env.fed_task_id ?: env.fed_id
+    myId = (artifact == 'koji-build') ? taskid : env.fed_rev
+    myNvr = env.nvr ?: ""
+    myCategory = (artifact == 'koji-build') ? 'functional' : 'static-analysis'
```

Comment:
@thrix please ack/nack the category. I got these looking at other examples, but we can obviously change them.

Reply:
This is ok for the pipeline